### DISCUSSION TITLE
How to make javascript in edge extension unreadable?

### DISCUSSION THREAD

**Discussion by timewalker08** [microsoft/MicrosoftEdge-Extensions#88] [Discussions]
Hi, Microsoft edge extension doesn't allow obfuscated javascript code. How you guys protect your javascript code?
Following is what my javascript code looks like:
`
var allGoodsInfo = {};

const refreshGoodsBtn = document.getElementById("btnRefreshGoods");
refreshGoodsBtn.onclick = function (e) {
    refreshGoods();
    //collectGoods();
};`

But I find some other edge extension javascript code looks like following:
`(()=>{
    "use strict";
    var t, e = {
        91942: (t,e,n)=>{
            n.d(e, {
                AFY: ()=>X,
                Alt: ()=>nt,
                BgB: ()=>xt,
                Bzc: ()=>Tt,
                Cr2: ()=>Vt,
                Csj: ()=>P,
                D2c: ()=>Yt,
                EOg: ()=>ie,
                EpL: ()=>J,
                Eye: ()=>Y,
                FMD: ()=>Lt,
                Fcr: ()=>Ce,
                G07: ()=>Zt,
                GPh: ()=>St,
                GjQ: ()=>M,
                H$A: ()=>et,
                HC1: ()=>C,
                HcK: ()=>yt,
                Hpx: ()=>ne,
                ILo: ()=>H,`

How to make javascript code unreadable like above code?

**Comment by sivsouvamMSFT** [comment]
Hello @timewalker08, thank you for reaching out to us. We will check it with our engineering team & revert.

**Comment by timewalker08** [comment]
Hi Siv Souvam,

Thanks for your reply. Looking forward to your help.

Thanks
Best regards

On May 16, 2023, at 18:28, Siv Souvam ***@***.***> wrote:



Hello @timewalker08<https://github.com/timewalker08>, thank you for reaching out to us. We will check it with our engineering team & revert.

**Comment by sivsouvamMSFT** [comment]
Hello @timewalker08,

As per our store policies, obfuscation is not allowed. The code has to be readable/reviewable as per policy. 

Please find more about the policy here: [Microsoft Edge Add-ons store developer policies](https://learn.microsoft.com/en-us/microsoft-edge/extensions-chromium/store-policies/developer-policies#117-obfuscation).

**Reply by timewalker08** [reply]
Hi @sivsouvamMSFT , right, I know obfuscation code is not allowed.
But, I do find that there are some extension whose javascript code is obfuscated.
Like this one: https://microsoftedge.microsoft.com/addons/detail/%E5%88%92%E8%AF%8D%E7%BF%BB%E8%AF%91/oikmahiipjniocckomdccmplodldodja
The javascript code in p.js likes:

(()=>{
    "use strict";
    var t, e = {
        91942: (t,e,n)=>{
            n.d(e, {
                AFY: ()=>X,
                Alt: ()=>nt,
                BgB: ()=>xt,
                Bzc: ()=>Tt,
                Cr2: ()=>Vt,
                Csj: ()=>P,
                D2c: ()=>Yt,
                EOg: ()=>ie,
                EpL: ()=>J,
                Eye: ()=>Y,
                FMD: ()=>Lt,
                Fcr: ()=>Ce,
                G07: ()=>Zt,
                GPh: ()=>St,
                GjQ: ()=>M,
                H$A: ()=>et,
                HC1: ()=>C,
                HcK: ()=>yt,
                Hpx: ()=>ne,
                ILo: ()=>H,
                IRm: ()=>k,
                IYJ: ()=>A,
                J4_: ()=>N,
                KZz: ()=>kt,
                KpW: ()=>je,

**Reply by sivsouvamMSFT** [reply]
Thank you @timewalker08 for sharing it. 

Please allow us some time to look into it & revert.
